src/components/BookList.js

```diff
@@ -1,45 +1,71 @@
-import { FcFullTrash } from 'react-icons/fc';
 import { useSelector, useDispatch } from 'react-redux';
+import { FcFullTrash } from 'react-icons/fc';
 import { removeBook } from '../redux/books/booksSlice';
 import './AddBook.css';
 import ChapterProgress from './ChapterProgress';
 import ChapterUpdateProgress from './ChapterUpdateProgress';
 
-const BooksList = () => {
+function BookList() {
+  const { books, isLoading, errorMsg } = useSelector(
+    (state) => state.books || [],
+  );
   const dispatch = useDispatch();
-  const books = useSelector((state) => state.books);
+
+  if (isLoading) {
+    return (
+      <div className="loading">
+        <h1>Loading...</h1>
+      </div>
+    );
+  }
+
+  if (errorMsg) {
+    return (
+      <div className="loading">
+        <h1>Failed to fetch ...</h1>
+      </div>
+    );
+  }
```

Comment:
![image](https://github.com/Benawi/Microverse-React-Bookstore/assets/93039359/00c34114-b613-4766-9f89-4591ce00d750)

- [ ] It would be nice if you use useEffect hook in your project to follow the project requirements 👍